src/server/game/Achievements/AchievementMgr.cpp

```diff
@@ -65,7 +65,7 @@ bool AchievementMgr::CanUpdateCriteriaTree(Criteria const* criteria, CriteriaTre
 {
     AchievementEntry const* achievement = tree->Achievement;
     if (!achievement)
-        return false;
+        return true;
```

Comment:
Take a look at Player::UpdateCriteria, it sends updates to the achievement manager, scenario, and guild system, which are all separate criteria handlers.